### DISCUSSION TITLE
Own Topography

### DISCUSSION THREAD

**Discussion by anneschulz1991** [gempy-project/gempy#731] [Q&A]
Hello everyone, I need your help! Is there a way to import your own topography in gempy? So similar to the geological units with XYZ coordinates. So far I have only seen that you can use a TIF file or that you can create an arbitrary topography from gempy. Can someone help me? Many greetings Anne

**Comment by AlexanderJuestel** [comment]
Hello @anneschulz1991,
You could save your data as tif file using the rasterio library or the save to tif function implemented in GemGIS.

Let me know if I can be of any more help :)

**Comment by anneschulz1991** [comment]
Hello Alexander, I have tried to implement your suggestion. Unfortunately, I did not succeed in creating a grid from my point coordinates. My file looks like this:

X,Y,Z
4364557.1426563300,5678329.9754381403,430.9512630000
4364558.1426563300,5678329.9754381403,430.9512630000
4365100.3757597785,5678870.4945186004,481.1981200000
4365101.3757597785,5678870.4945186004,481.1981200000
4365643.6088632261,5679411.0135990595,425.2071840000
4365644.6088632261,5679411.0135990595,425.2071840000
4366186.8419666747,5679951.5326795196,370.1652220000
4366187.8419666747,5679951.5326795196,370.1652220000
4366730.0750701223,5680492.0517599788,465.0390010000
4366731.0750701223,5680492.0517599788,465.0390010000
4367273.3081735708,5681032.5708404388,350.7116090000
4367274.3081735708,5681032.5708404388,350.7116090000
4367816.5412770184,5681573.0899208980,328.9978640000
4367817.5412770184,5681573.0899208980,328.9978640000
4368359.7743804669,5682113.6090013580,272.7081910000
4368360.7743804669,5682113.6090013580,272.7081910000
4368903.0074839145,5682654.1280818172,329.0356750000
4368904.0074839145,5682654.1280818172,329.0356750000
4369446.2405873630,5683194.6471622773,394.3047180000
4369447.2405873630,5683194.6471622773,394.3047180000
4369989.4736908115,5683735.1662427373,412.1142580000
4369990.4736908115,5683735.1662427373,412.1142580000
4370532.7067942591,5684275.6853231965,338.8646240000
4370533.7067942591,5684275.6853231965,338.8646240000
4371075.9398977077,5684816.2044036565,325.6873780000
4371076.9398977077,5684816.2044036565,325.6873780000
4371619.1730011553,5685356.7234841157,314.3685610000
4371620.1730011553,5685356.7234841157,314.3685610000
4372162.4061046029,5685897.2425645757,289.6987000000
4372163.4061046029,5685897.2425645757,289.6987000000
4372705.6392080504,5686437.7616450349,389.0102840000
4372706.6392080504,5686437.7616450349,389.0102840000
4373248.8723114990,5686978.2807254950,469.0675350000
4373249.8723114990,5686978.2807254950,469.0675350000
4373792.1054149466,5687518.7998059541,449.9636540000
4373793.1054149466,5687518.7998059541,449.9636540000
4374335.3385183951,5688059.3188864142,453.2296750000
4374336.3385183951,5688059.3188864142,453.2296750000
4374878.5716218427,5688599.8379668733,498.1649780000
4374879.5716218427,5688599.8379668733,498.1649780000
4375421.8047252912,5689140.3570473334,468.3771670000
4375422.8047252912,5689140.3570473334,468.3771670000
4375965.0378287397,5689680.8761277935,438.8978880000
4375966.0378287397,5689680.8761277935,438.8978880000
4376508.2709321873,5690221.3952082526,429.5755000000
4376509.2709321873,5690221.3952082526,429.5755000000
4377051.5040356359,5690761.9142887127,416.7854310000
4377052.5040356359,5690761.9142887127,416.7854310000
4377594.7371390834,5691302.4333691718,435.8194270000
4377595.7371390834,5691302.4333691718,435.8194270000
4378137.9702425320,5691842.9524496319,470.2549440000
4378138.9702425320,5691842.9524496319,470.2549440000
4378681.2033459796,5692383.4715300910,474.3962400000
4378682.2033459796,5692383.4715300910,474.3962400000
4379224.4364494281,5692923.9906105511,478.3889160000
4379225.4364494281,5692923.9906105511,478.3889160000
4379767.6695528757,5693464.5096910102,461.3367920000
4379768.6695528757,5693464.5096910102,461.3367920000
4380310.9026563233,5694005.0287714703,342.7093510000
4380311.9026563233,5694005.0287714703,342.7093510000
4380854.1357597718,5694545.5478519304,329.0068360000
4380855.1357597718,5694545.5478519304,329.0068360000
4381397.3688632194,5695086.0669323895,327.8271480000
4381398.3688632194,5695086.0669323895,327.8271480000
4381940.6019666679,5695626.5860128496,334.6157230000
4381941.6019666679,5695626.5860128496,334.6157230000
4382483.8350701155,5696167.1050933087,339.6142880000
4382484.8350701155,5696167.1050933087,339.6142880000
4383027.0681735640,5696707.6241737688,343.4789120000
4383028.0681735640,5696707.6241737688,343.4789120000
4383570.3012770116,5697248.1432542279,342.1058650000
4383571.3012770116,5697248.1432542279,342.1058650000
4384113.5343804602,5697788.6623346880,346.7103880000
4384114.5343804602,5697788.6623346880,346.7103880000
4384656.7674839078,5698329.1814151471,368.6368100000
4384657.7674839078,5698329.1814151471,368.6368100000
4385200.0005873563,5698869.7004956072,383.9331360000
4385201.0005873563,5698869.7004956072,383.9331360000
4385743.2336908048,5699410.2195760673,369.0009160000
4385744.2336908048,5699410.2195760673,369.0009160000
4386286.4667942524,5699950.7386565264,348.7060850000
4386287.4667942524,5699950.7386565264,348.7060850000
4386829.6998977009,5700491.2577369865,376.6770020000
4386830.6998977009,5700491.2577369865,376.6770020000
4387372.9330011485,5701031.7768174456,419.0807190000
4387373.9330011485,5701031.7768174456,419.0807190000
4387916.1661045961,5701572.2958979057,470.5421140000
4387917.1661045961,5701572.2958979057,470.5421140000
4388459.3992080437,5702112.8149783649,507.0823060000
4388460.3992080437,5702112.8149783649,507.0823060000
4389002.6323114922,5702653.3340588249,498.1152950000
4389003.6323114922,5702653.3340588249,498.1152950000
4389545.8654149398,5703193.8531392841,498.0458370000
4389546.8654149398,5703193.8531392841,498.0458370000
4390089.0985183883,5703734.3722197441,502.4564510000
4390090.0985183883,5703734.3722197441,502.4564510000
4390632.3316218359,5704274.8913002033,458.0641480000
4390633.3316218359,5704274.8913002033,458.0641480000
4391175.5647252845,5704815.4103806634,368.6204530000
4391176.5647252845,5704815.4103806634,368.6204530000
4391718.7978287330,5705355.9294611234,329.9288640000
4391719.7978287330,5705355.9294611234,329.9288640000
4392262.0309321806,5705896.4485415826,314.7163700000
4392263.0309321806,5705896.4485415826,314.7163700000
4392805.2640356291,5706436.9676220426,307.6549680000
4392806.2640356291,5706436.9676220426,307.6549680000
4393348.4971390767,5706977.4867025018,288.4982600000
4393349.4971390767,5706977.4867025018,288.4982600000
4393891.7302425252,5707518.0057829618,290.0797120000
4393892.7302425252,5707518.0057829618,290.0797120000
4394434.9633459728,5708058.5248634210,349.1842040000
4394435.9633459728,5708058.5248634210,349.1842040000
4394978.1964494213,5708599.0439438811,335.2706300000
4394979.1964494213,5708599.0439438811,335.2706300000
4395521.4295528689,5709139.5630243402,287.5362850000
4395522.4295528689,5709139.5630243402,287.5362850000
4396064.6626563165,5709680.0821048003,256.8961790000
4396065.6626563165,5709680.0821048003,256.8961790000
4396607.8957597651,5710220.6011852603,239.1199800000
4396608.8957597651,5710220.6011852603,239.1199800000
4397151.1288632127,5710761.1202657195,229.0055240000
4397152.1288632127,5710761.1202657195,229.0055240000
4397694.3619666612,5711301.6393461796,225.6679690000
4397695.3619666612,5711301.6393461796,225.6679690000
4398237.5950701088,5711842.1584266387,233.6241610000
4398238.5950701088,5711842.1584266387,233.6241610000
4398780.8281735573,5712382.6775070988,243.2461090000
4398781.8281735573,5712382.6775070988,243.2461090000
4399324.0612770049,5712923.1965875579,242.9788210000
4399325.0612770049,5712923.1965875579,242.9788210000
4399867.2943804534,5713463.7156680180,236.5364990000
4399868.2943804534,5713463.7156680180,236.5364990000
4400410.5274839010,5714004.2347484771,238.5351870000
4400411.5274839010,5714004.2347484771,238.5351870000
4400953.7605873495,5714544.7538289372,268.8639220000
4400954.7605873495,5714544.7538289372,268.8639220000
4401496.9936907981,5715085.2729093973,287.0143430000
4401497.9936907981,5715085.2729093973,287.0143430000
4402040.2267942457,5715625.7919898564,273.1029050000
4402041.2267942457,5715625.7919898564,273.1029050000
4402583.4598976942,5716166.3110703165,282.6657410000
4402584.4598976942,5716166.3110703165,282.6657410000
4403126.6930011418,5716706.8301507756,300.9350590000
4403127.6930011418,5716706.8301507756,300.9350590000
4403669.9261045894,5717247.3492312357,290.6591190000
4403670.9261045894,5717247.3492312357,290.6591190000
4404213.1592080370,5717787.8683116948,279.9266970000
4404214.1592080370,5717787.8683116948,279.9266970000
4404756.3923114855,5718328.3873921549,287.0336910000
4404757.3923114855,5718328.3873921549,287.0336910000
4405299.6254149331,5718868.9064726140,281.9020080000
4405300.6254149331,5718868.9064726140,281.9020080000
4405842.8585183816,5719409.4255530741,276.0702820000
4405843.8585183816,5719409.4255530741,276.0702820000
4406386.0916218292,5719949.9446335332,307.9152830000
4406387.0916218292,5719949.9446335332,307.9152830000
4406929.3247252777,5720490.4637139933,362.4359130000
4406930.3247252777,5720490.4637139933,362.4359130000
4407472.5578287262,5721030.9827944534,435.9484860000
4407473.5578287262,5721030.9827944534,435.9484860000
4408015.7909321738,5721571.5018749125,482.5358280000
4408016.7909321738,5721571.5018749125,482.5358280000
4408559.0240356224,5722112.0209553726,520.0455320000
4408560.0240356224,5722112.0209553726,520.0455320000
4409102.2571390700,5722652.5400358317,562.8092650000
4409103.2571390700,5722652.5400358317,562.8092650000
4409645.4902425185,5723193.0591162918,576.7267460000
4409646.4902425185,5723193.0591162918,576.7267460000
4410188.7233459661,5723733.5781967510,580.3631590000
4410189.7233459661,5723733.5781967510,580.3631590000
4410731.9564494146,5724274.0972772110,578.0328370000
4410732.9564494146,5724274.0972772110,578.0328370000
4411275.1895528622,5724814.6163576702,563.1813960000
4411276.1895528622,5724814.6163576702,563.1813960000
4411818.4226563098,5725355.1354381302,546.3802490000
4411819.4226563098,5725355.1354381302,546.3802490000

I have created a geopackage file from it. Using the Z-values I tried to create a raster with "Grid(nearest neighbor)". Unfortunately, QGIS shows me that it should have worked but the grid is not visible and a wrong maximum and minimum is displayed. Can you tell me what the problem is? Thanks in advance for your help!
Many greetings
Anne

**Comment by Japhiolite** [comment]
@anneschulz1991 
I had a look in qgis at the csv you provided, it looks like a very elongated grid, like always 2 points next to each other in the x direction?
So I assume you're looking at a kinda 2D model? (or 2.5 Dimensions)
![image](https://user-images.githubusercontent.com/13364692/197990423-68d61db8-ca72-451a-97d0-f758399aa3ad.png)

**Comment by anneschulz1991** [comment]
Yes, that is exactly the plan. I think I could now create all the important files for it. Thanks again for your help. But I still have one question! Is there a possibility to specify a higher resolution than 100 in all three spatial directions?
Many greetings
Anne

**Reply by Japhiolite** [reply]
Great to hear!
Concerning resolution: In Principle yes, but the bottleneck will be memory of your hardware. The surfaces themselves are not bound to the resolution of the regular grid, and there has been work on using the surfaces as input in other discretization software. But there I don't know much more than that it was done...
